docs/octopus_deploy_smtp_config.md

```diff
@@ -0,0 +1,37 @@
+# octopus_deploy_smtp_config
+
+Tests the server-wide SMTP configuration. This can be found in the UI under Configuration -> SMTP
+
+## Example
+
+```ruby
+describe octopus_deploy_smtp_config do
+  it { should be_configured }
+  it { should have_from_address('hello@my.email.domain') }
+  it { should be_using_credentials('myusername') }
+  it { should_not be_using_credentials('myspecialusername') }
+  it { should be_on_port(25) }
+  it { should be_on_host('smtp.my.email.domain')}
+  it { should be_using_ssl }
+end
+```
+
+#### Type
```

Comment:
Not sure what this block is telling me...

Reply:
Basically the different ways you can construct to object. Everything will have a shorter alias going forward, and auth info is optional. So there are four ways you can construct any given type